netobserv/v1alpha1/flowcollector_types.go

```diff
@@ -0,0 +1,436 @@
+}
+
+// FlowCollectorFLP defines the desired flowlogs-pipeline state of FlowCollector
+type FlowCollectorFLP struct {
+	// Important: Run "make generate" to regenerate code after modifying this file
+
+	//+kubebuilder:validation:Enum=DaemonSet;Deployment
+	//+kubebuilder:default:=DaemonSet
+	// Kind is the workload kind, either DaemonSet or Deployment
+	// When using Kafka, this option only affects the flowlogs-pipeline ingester, not the transformer.
+	Kind string `json:"kind,omitempty"`
+
+	//+kubebuilder:validation:Minimum=0
+	//+kubebuilder:default:=1
+	// Replicas defines the number of replicas (pods) to start for Deployment kind. Ignored for DaemonSet.
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// HPA spec of an horizontal pod autoscaler to set up for the collector Deployment. Ignored for DaemonSet.
+	// +optional
+	HPA *FlowCollectorHPA `json:"hpa,omitempty"`
+
+	//+kubebuilder:validation:Minimum=1025
+	//+kubebuilder:validation:Maximum=65535
+	//+kubebuilder:default:=2055
+	// Port is the collector port: either a service port for Deployment kind, or host port for DaemonSet kind
+	// By conventions, some value are not authorized port must not be below 1024 and must not equal this values:
+	// 4789,6081,500, and 4500
+	Port int32 `json:"port,omitempty"`
+
+	//+kubebuilder:validation:Minimum=1
+	//+kubebuilder:validation:Maximum=65535
+	//+kubebuilder:default:=8080
+	// HealthPort is a collector HTTP port in the Pod that exposes the health check API
+	HealthPort int32 `json:"healthPort,omitempty"`
+
+	//+kubebuilder:validation:Minimum=1
+	//+kubebuilder:validation:Maximum=65535
+	//+kubebuilder:default:=9090
+	// PrometheusPort is the prometheus HTTP port: this port exposes prometheus metrics
+	PrometheusPort int32 `json:"prometheusPort,omitempty"`
+
+	//+kubebuilder:default:="quay.io/netobserv/flowlogs-pipeline:main"
+	// Image is the collector image (including domain and tag)
+	Image string `json:"image,omitempty"`
+
+	//+kubebuilder:validation:Enum=IfNotPresent;Always;Never
+	//+kubebuilder:default:=IfNotPresent
+	// ImagePullPolicy is the Kubernetes pull policy for the image defined above
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
+
+	//+kubebuilder:validation:Enum=trace;debug;info;warn;error;fatal;panic
+	//+kubebuilder:default:=info
+	// LogLevel defines the log level for the collector runtime
+	LogLevel string `json:"logLevel,omitempty"`
+
+	//+kubebuilder:default:={requests:{memory:"100Mi",cpu:"100m"},limits:{memory:"300Mi"}}
+	// Compute Resources required by this container.
+	// Cannot be updated.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,8,opt,name=resources"`
+
+	//+kubebuilder:default:=true
+	// EnableKubeProbes is a flag to enable or disable Kubernetes liveness/readiness probes
+	EnableKubeProbes bool `json:"enableKubeProbes,omitempty"`
```

Comment:
Reconsider this bool

Reply:
do you mean reconsider as "use an enum instead" or do you mean we shouldn't allow to disable kube probes?

Reply:
I meant use an enum yes, we don't allow bools, think I left a more thorough comment somewhere else